lib/hocon/impl/config_delayed_merge_object.rb

```diff
@@ -30,6 +32,121 @@ def initialize(origin, stack)
 
   attr_reader :stack
 
+  def new_copy(status, origin)
+    if status != resolve_status
+      raise Hocon::ConfigError::ConfigBugOrBrokenError.new(
+                "attempt to create resolved ConfigDelayedMergeObject")
+    end
+    Hocon::Impl::ConfigDelayedMergeObject.new(origin, @stack)
+  end
+
+  def resolve_substitutions(context, source)
+    merged = Hocon::Impl::ConfigDelayedMerge.resolve_substitutions(self, @stack, context, source)
+    merged.as_object_result
+  end
+
+  def make_replacement(context, skipping)
+    Hocon::Impl::ConfigDelayedMerge.make_replacement(context, @stack, skipping)
+  end
+
+  def resolve_status
+    Hocon::Impl::ResolveStatus::UNRESOLVED
+  end
+
+  def replace_child(child, replacement)
+    new_stack = Hocon::Impl::AbstractConfigValue.replace_child_in_list(@stack, child, replacement)
+    if new_stack == nil
+      nil
+    else
+      self.class.new(origin, new_stack)
+    end
+  end
+
+  def has_descendant?(descendant)
+    has_descendant_in_list?(@stack, descendant)
+  end
+
+  def relativized(prefix)
+    new_stack = []
+    @stack.each { |o|
+      new_stack << o.relativized(prefix)
+    }
+    self.class.new(origin, new_stack)
+  end
+
+  def ignores_fallbacks?
+    Hocon::Impl::ConfigDelayedMerge.stack_ignores_fallbacks?(@stack)
+  end
+
+  def merged_with_the_unmergeable(fallback)
+    require_not_ignoring_fallbacks
+
+    merged_stack_with_the_unmergeable(@stack, fallback)
+  end
+
+  def merged_with_object(fallback)
+    merged_with_non_object(fallback)
+  end
+
+  def merged_with_non_object(fallback)
+    require_not_ignoring_fallbacks
+
+    merged_stack_with_non_object(@stack, fallback)
+  end
+
+  # No implementation of withFallback here,
+  # just use the implementation in the super-class
```

Comment:
not possible, since there is another version of with fallback that just a different parameter type